src/main.cpp

```diff
@@ -33,6 +33,8 @@
 #include "utils/SettingsUtils.h"
```

Comment:
all includes: `codechandler/*` can be removed

do you think we can move part of `OpenStream` to Session to remove all includes of `samplereader/*` from main?

Reply:
there a bit of mess to be fixed also with all other includes:
example:
//#include "ADTSReader.h"
//#include "TSReader.h"
//#include "WebmReader.h"
//#include "aes_decrypter.h"
//#include "oscompat.h"
//#include "parser/DASHTree.h"
//#include "parser/HLSTree.h"
//#include "parser/SmoothTree.h"
//#include "utils/Base64Utils.h"
//#include "utils/SettingsUtils.h"
//#include "utils/StringUtils.h"
#include <chrono>
#include <algorithm>
are unused and probably also others

#include <string.h> can be changed to #include <string>
and miss to add:
#include <map>

Reply:
> do you think we can move part of OpenStream to Session to remove all includes of samplereader/* from main?

Yes but probably better for a separate PR. Looks like `CStream` would be a good place to handle this and use the factory pattern to create the readers (pass in rep to get flags/containertype). ~150 lines is too much for a single function